Absolutepath

szerdahelyir opened the pull request:
absolutepath done

szerdahelyir committed:
absolutepath

absolutepath kezdetleges

szerdahelyir committed:
Makefile upload

Github Desktop didn't upload it

szerdahelyir committed:
changes

absolute path, base class, bug fixes

szerdahelyir committed:
Update Makefile

szerdahelyir committed:
absolutepath pimp

wajzy commented on ancestor.cpp:
```diff
@@ -0,0 +1,8 @@
+#include "ancestor.h"
+
+ancestor::ancestor(string name, directory* parent)
```
Nem túl kifejező ez a név, bár valóban két osztály közös őse. Arra kellene utalni, hogy ez egy tetszőleges fájlrendszerbeli bejegyzést takarhat. Pl. item, vagy hasonló.

wajzy commented on ancestor.h:
```diff
@@ -0,0 +1,21 @@
+public:
+    ancestor(string,directory*);
+    virtual ~ancestor()=0;
+    virtual string getname()=0;
```
A getname, getparent tagfüggvényeket nem lehetne a közös ősben definiálni? Hirtelen nem látom, miben fog a két leszármazott működése eltérni egymástól.

wajzy commented on directory.cpp:
```diff
@@ -1,36 +1,40 @@
 }
 
-void directory::ls()
+list<ancestor *> directory::getSubdirectories() const
```
Ez a függvénynév is félrevezető, mert a fájlbejegyzések is a visszaadott értékek között szerepelnek, nem csak a könyvtárak.

wajzy commented on directory.cpp:
```diff
@@ -41,13 +45,24 @@ bool directory::hasDirs()
 {
     for(auto it=subdirectories.begin();it!=subdirectories.end();){
-        if((*it)->getName()==todelete &&!(*it)->hasDirs()){
+        directory* temp=dynamic_cast<directory*>(*it);
```
Ugye nem kizárlólag emiatt kellettek a virtuális függvények az ősbe?

wajzy commented on directory.h:
```diff
@@ -1,30 +1,26 @@
-    string name;
-    directory* parent;
-    list <directory*> subdirectories;
+    list<ancestor*> subdirectories;
```
Nem vagyok benne biztos, hogy a lista szerencsés választás. Szerintem egy map-pel gyorsabban meg lehetne találni egy konkrét bejegyzést, pl. törlés esetén.

wajzy commented on file.h:
```diff
@@ -0,0 +1,16 @@
+class file:public ancestor
+{
+private:
+    string content;
```
Mi használja ezt az adattagot?

wajzy commented on filesystem.cpp:
```diff
@@ -1,64 +1,222 @@
     return 0;
 }
 
+vector<string> filesystem::split(const string& str, const char& delim)
```
Lehet, hogy megy ez egyszerűbben is: https://www.geeksforgeeks.org/tokenizing-a-string-cpp/

wajzy commented on filesystem.cpp:
```diff
@@ -1,64 +1,222 @@
+    }
+}
+
+int filesystem::absolutepathmkdir(string path){
```
Itt egymás után több függvénynek is az eleje majdnem tökéletesen ugyanaz. Ezeket a redundáns részeket el kellene tüntetni.

wajzy commented on test.txt:
```diff
@@ -0,0 +1,8 @@
+touch file
+cd..
+ls
+exit
```
Ez hogyan teszteli az abszolút elérési útvonalakat?

wajzy reviewed (changes requested):
Ez a dynamiccast-os trükk nyilván működik az ősosztályt címző mutató által kijelölt objektum típusának ellenőrzésére, de nem tudom, mennyire hatékony megoldást jelent. Én valószínűleg inkább készítettem volna egy bool isFile()-szerű tagfüggvényt az ősbe, ami az egyik leszármazottban igazzal, a másikban hamissal tér vissza. 
Ami az abszolút elérési utakat illeti, az ránézésre megvan, de a sok redundáns részt mindenképpen el kellene tüntetni.

hegyhati commented:
> Ez a dynamiccast-os trükk nyilván működik az ősosztályt címző mutató által kijelölt objektum típusának ellenőrzésére, de nem tudom, mennyire hatékony megoldást jelent. Én valószínűleg inkább készítettem volna egy bool isFile()-szerű tagfüggvényt az ősbe, ami az egyik leszármazottban igazzal, a másikban hamissal tér vissza.

Itt vitatkoznek, szerintem jobb a `dynamic_cast`, az ososztalynak ha nem muszaj, ne tudjon arrol, milyen gyerekei vannak. Szoval szerintem inkabb az maradjon.

szerdahelyir committed:
javasolt hibák kijavítása

függvények/adattagok neveinek átírása, getparent és getname fgv-ek kifejtése ősosztályba, új splitelő függvény, redundancia kezelése

szerdahelyir committed:
makefile fix

szerdahelyir committed:
makefile and circleci fix v2

szerdahelyir committed:
circleci fix v3

szerdahelyir committed:
circleci fix v4

szerdahelyir commented:
@wajzy külön függvénybe raktam a redundáns tartalmakat ahol lehetett, javítottam a neveken, és a tokenizáló függvényt is lecseréltem, viszont a map helyett bent hagytam a listás megoldást mert minden további funkció(echo, fájlba kimentés)-t listával oldottunk meg

szerdahelyir committed:
conflig resolving

szerdahelyir committed:
ancestor remove

szerdahelyir committed:
Merge branch 'master' into absolutepath

hegyhati commented:
@wajzy egyszer mar approve-oltad, azota felment meg 2 commit (+1 update).  Nezd meg pls, hogy tovabbra is aldasod-e a dolgon.

szerdahelyir commented:
@hegyhati csak conflict resolve miatt volt az a 2 commit, változtatások nem történtek
